main.tf

```diff
@@ -47,46 +47,61 @@ module "kms" {
 # ---------------------------------------------------------------------------------------------------------------------
 module "aws_eks" {
   source  = "terraform-aws-modules/eks/aws"
-  version = "v17.20.0"
-
-  create_eks      = var.create_eks
-  manage_aws_auth = false
-
-  cluster_name          = var.cluster_name == "" ? module.eks_tags.id : var.cluster_name
-  cluster_version       = var.kubernetes_version
-  cluster_iam_role_name = local.cluster_iam_role_name
-
-  # NETWORK CONFIG
-  vpc_id  = var.vpc_id
-  subnets = var.private_subnet_ids
-
-  cluster_endpoint_private_access                = var.cluster_endpoint_private_access
-  cluster_create_endpoint_private_access_sg_rule = local.cluster_create_endpoint_private_access_sg_rule
-  cluster_endpoint_private_access_cidrs          = local.cluster_endpoint_private_access_cidrs
-  cluster_endpoint_private_access_sg             = local.cluster_endpoint_private_access_sg
-
+  version = "v18.10.0"
+  create  = var.create_eks
+
+  cluster_name     = var.cluster_name == "" ? module.eks_tags.id : var.cluster_name
+  cluster_version  = var.cluster_version
+  cluster_timeouts = var.cluster_timeouts
+
+  # IAM Role
+  iam_role_use_name_prefix      = false
+  iam_role_name                 = local.cluster_iam_role_name
+  iam_role_path                 = var.iam_role_path
+  iam_role_permissions_boundary = var.iam_role_permissions_boundary
+  iam_role_additional_policies  = var.iam_role_additional_policies
+
+  # EKS Cluster VPC Config
+  subnet_ids                           = var.private_subnet_ids
+  cluster_endpoint_private_access      = var.cluster_endpoint_private_access
   cluster_endpoint_public_access       = var.cluster_endpoint_public_access
   cluster_endpoint_public_access_cidrs = var.cluster_endpoint_public_access_cidrs
 
-  worker_create_security_group  = var.worker_create_security_group
-  cluster_log_retention_in_days = var.cluster_log_retention_in_days
+  # Kubernetes Network Config
+  cluster_ip_family         = var.cluster_ip_family
+  cluster_service_ipv4_cidr = var.cluster_service_ipv4_cidr
+
+  # Cluster Security Group
+  create_cluster_security_group           = true
```

Comment:
Should we move the hardcoded true(s) as variables? do we see any use case or scenarios where users may want to define this as false?

Reply:
We don't want this value to be modified by the user.  I kept it for information purpose in the module which explains what all its creating but hiding from the user to modify the values.